modules/migrate.py

```diff
@@ -0,0 +1,485 @@
+
+Example command lines:
+    migrate.py -c migrate.csv
+    migrate.py -c migrate.csv -s
+"""
+
+import argparse
+import csv
+import glob
+import json
+import logging
+import os
+import re
+import subprocess
+
+from bs4 import BeautifulSoup
+
+from s9logging import s9logging
+import svn.project_svn as svn
+
+parser = argparse.ArgumentParser(description='Migrate content from a widget to '
+    'modularized form of the same widget in a given project.')
+parser.add_argument('-c', '--config', required=True, help='CSV file specifying '
+    'widgets migrating to modules in each project, each row in the form "'
+    'repo_shortname,environment,widget_directory_name,module_directory_name"')
+parser.add_argument('-s', '--skip-commit', action='store_true', default=False,
+    help='Whether to skip svn commit of project changes')
+
+s9logging.configureLogging()
+log = logging.getLogger(__name__)
+
+def _getSpecsFromCsv():
+    """Returns a list of tuples of the form (source name, source environment,
+    widget directory name, module directory name) taken from the CSV
+    configuration file.
+    """
+    results = []
+    if args.config:
+        with open(args.config, 'rb') as f:
+            reader = csv.reader(f)
+            try:
+                for row in reader:
+                    # Strip any whitespace in row contents.
+                    row = map(str.strip, row)
+                    if len(row) < 4 or not (row[0] and row[1] and row[2] and
+                            row[3]):
+                        log.warning('CSV has invalid number of arguments at '
+                            'line %s, skipping line.\n', reader.line_num)
+                    else:
+                        results.append((row[0], row[1], row[2], row[3]))
+            except csv.Error as e:
+                log.error('Unable to parse CSV at line %s, skipping rest of'
+                    ' file.\n', reader.line_num)
+
+    return results
+
+
+def _getAllHTMLFiles(rootDir):
+    """Returns a list of all HTML files under the root directory.
+
+    Args:
+        rootDir - String absolute path of the directory tree to search.
+    """
+    htmlFiles = []
+    for root, dirs, files in os.walk(rootDir):
+        for filename in [f for f in files if f.endswith('.html')]:
+            htmlFiles.append(os.path.join(root, filename))
+    return htmlFiles
+
+def _updateHTMLFile(filePath, widgetAbsolutePath, modularWidgetAbsolutePath):
+    """Updates file and linked widget JSON config files.
+
+    Updates the specified HTML file changing all references to the
+    non-modular widget to instead reference the modular widget, and all paths
+    relative to the non-modular widget to be relative to the modular widget.
+    Also fixes any JSON configuration files references from a widget's <object>
+    tag.
+    """
+    log.debug('Reading HTML file: %s', filePath)
+    with open(filePath, 'rb') as htmlFile:
+        lines = ''.join(htmlFile.readlines())
```

Comment:
Since we are joining all the lines, it might be clearer to rename `lines` to `htmlContent`. Makes it a bit clearer below that it isn't an array of strings.